### DISCUSSION TITLE
[ SOLVED ] - Try latest GIT-version! Problems with exception 6 on ESP32_Relay X4_v1.1 - ESP32-WROOM

### DISCUSSION THREAD

**Discussion by balou1974** [arendst/Tasmota#17285] [Q&A] [answered]
Hello,

I got a problem with a relay-card from aliexpress ESP32_Relay X4_v1.1 - it is working with default options (1 prog. button, 1 prog. led and 4x prog. relays) but as soon as I try to add a rf 433mhz receiver the whole thing reboots and clears the setting.

I tried different gpios and different power sources (the relay card can use 85-230V ac or 5 to 30V DC). I think this a flash or interrupt problem but kc868-a4 works with the same firmware (Tasmota 12.0.2.6 - self compiled and precompiled tested).

As soon as I connect the data-pin to the selected gpio it is over and the ESP32  is bringing an exception 6, reboots and clears the
settings.

GPIO0 Button, GPIO23 led, GPIO25 Relay 4, GPIO26 Relay 3, GPIO32 Relay 1, GPIO33 Relay 2

Tried different GPIOs for RFrecv GPIO 2, 27, 22, 35 nothing is working ... 

```
05:09:52.008 HTP: Web-Server aktiv bei tasmota-FFA12C-0300 mit IP-Adresse 192.168.178.126
05:09:52.012 RSL: INFO1 = {"Info1":{"Module":"ESP32-DevKit","Version":"12.2.0.6(tasmota)","FallbackTopic":"cmnd/DVES_FFA12C_fb/","GroupTopic":"cmnd/tasmotas/"}}
05:09:52.025 RSL: INFO2 = {"Info2":{"WebServerMode":"Admin","Hostname":"tasmota-FFA12C-0300","IPAddress":"192.168.178.126","IP6Global":"","IP6Local":"fe80::c249:efff:feff:a12c"}}
05:09:52.037 RSL: INFO3 = {"Info3":{"RestartReason":{"Exception":6,"Reason":"IntegerDivideByZero","EPC":"4008163e","EXCVADDR":"00000000","CallChain":["4008163b"]},"BootCount":42}}
05:09:56.979 QPC: Reset
```

**Comment by barbudor** [comment]
Does the problem occurs when you connect the Rfrecv on a GPIO with is NOT set as Rfrecv?
If yes, the problem is electric
If not, please get the logs from the serial (use a DC power source to avoid any AC risk) with SerialLog 3

**Comment by balou1974** [comment]
Thank you for the reply!

The problem will occur if I connect the data line from the RF receiver to the configured RFrecv gpio / nothing will happen if I configure the gpio as IRrecv

**Reply by barbudor** [reply]
That was not my question
I ask if the exception occurs if you connect the RF receiver is connected to a GPIO not configured as rfrecv

**Reply by balou1974** [reply]
As far as I know the problem will only occur if I connect the RF Receiver to a gpio configured as RFrecv. I tried free gpios (no function asigned) and everything is ok - no reset/exception. The problem will only occur if I connect the RF receiver to a gpio asigned as RFrecv.

**Reply by barbudor** [reply]
Thanks
That seems to confirm a problem in the Rfrecv software

**Reply by balou1974** [reply]
Yes, It seems like a bug ...

**Comment by balou1974** [comment]
Today, I received a new Kincony KC868-A4 Rev. 1.2.4 and what should I say? There is the same problem, the default port for RFrecv is GPIO19 as soon as I active it the will be an exception:

```
00:00:02.900 RSL: INFO1 = {"Info1":{"Module":"KC868-A4","Version":"12.2.0.6(bluetooth)","FallbackTopic":"cmnd/DVES_4040E4_fb/","GroupTopic":"cmnd/tasmotas/"}}
00:00:02.913 RSL: INFO2 = {"Info2":{"WebServerMode":"Admin","Hostname":"tasmota-4040E4-0228","IPAddress":"192.168.178.127","IP6Global":"","IP6Local":"fe80::b68a:aff:fe40:40e4"}}
00:00:02.925 RSL: INFO3 = {"Info3":{"RestartReason":{"Exception":6,"Reason":"IntegerDivideByZero","EPC":"4008149e","EXCVADDR":"00000000","CallChain":["4008149b"]},"BootCount":21}}
```

**Reply by balou1974** [reply]
What wonders me is, my second, slightly older KC868-A4 (same Revision) works fine with the same settings and same firmware.

**Reply by balou1974** [reply]
I think this could be a problem with the configuration files / maybe some older firmware prepared other files / settings than the newer one? Tasmota-settings are kept between the firmware-updates. The second KC868-A4 was flashed with an older tasmota version at the start and was updated step-by-step ...

**Comment by balou1974** [comment]
I can confirm that the newer KC868-A4 will work fine if I restore the complete settings from the older/working unit to the new unit!

Here is what I did:
- Backup working unit to Config_tasmota_xxxxxx_xxx_12.2.0.6.dmp
- Restore Config_tasmota_xxxxxx_xxx_12.2.0.6.dmp (from working unit) to faulty unit
- FAULTY UNIT WORKS !!!

```
21:16:09.176 RSL: RESULT = {"Time":"2022-12-08T21:16:09","RfReceived":{"Data":"0x544551","Bits":24,"Protocol":1,"Pulse":329}}
21:16:10.303 RSL: RESULT = {"Time":"2022-12-08T21:16:10","RfReceived":{"Data":"0x54555F","Bits":24,"Protocol":1,"Pulse":328}}
21:16:14.636 RSL: RESULT = {"Time":"2022-12-08T21:16:14","RfReceived":{"Data":"0x544551","Bits":24,"Protocol":1,"Pulse":328}}
```

**Reply by balou1974** [reply]
So this must be a bug creating initial firmware settings in newer RFrecv enabled tasmota firmware.

**Reply by barbudor** [reply]
Would you agree to share a DMP of the working and non working unit so we can compare?
If you prefer you can contact me privately on discord

**Comment by arendst** [comment]
I've looked at the RCSwitch code. There are not many divides in the code so I think the issue is caused by a detected protocol having a PreambleFactor of 0.

If you compile your own code pls change file ``lib/lib_rf/rc-switch/src/RCSwitch.cpp`` line 761 from:
```
    } else {
```
into:
```
    } else if (pro.PreambleFactor > 0) {
```
and retry.

**Comment by arendst** [comment]
I just did a test on an ESP8266 and it immediatly restarts with exceptions 0.

After the above fix is installed it's stable.

Just merged the fix in latest development branch.

**Reply by balou1974** [reply]
@barbudor: I can share faulty and working dumps, but it should be privatly done - sorry for the late reply and thanks for your help!
@arendst: I will build a new version and try it - thank you a lot for you great work !!!

**Comment by balou1974** [comment, marked as answer]
I tried the latest GIT-source, compiled it, flashed it - and I can say it is working again!!!

Thanks a lot!!!
